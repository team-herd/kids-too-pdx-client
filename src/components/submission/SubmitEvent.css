.Form {
  display: block;
  padding: .5em;
  legend {
    text-align: center;
  }
  input {
    width: 100%;
  }
  textarea {
    width: 100%;
  }
  fieldset {
    margin: .5em;
  }
  button {
    margin-left: 40%;
    margin-right: 40%;
    padding: 1em;
    border-radius: 5px;
    background-color: white;
  }

}