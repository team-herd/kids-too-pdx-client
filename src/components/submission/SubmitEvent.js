import React from 'react';
import styles from './SubmitEvent.css';

function SubmitEvent() {
  return (
    <>
      <form className={styles.Form}>
        <fieldset>
          <legend>Contact Info</legend>

          <label name="name">Name:
            <input type="text" />
          </label>

          <label name="email">Email:
            <input type="email" />
          </label>

          <label>Phone #:
            <input type="tel" />
          </label>
        </fieldset>
        <fieldset>
          <legend>Event Info</legend>

          <label>Name:
            <input type="text" />
          </label>

          <label>Date:
            <input type="date" />
          </label>

          <label>Time:
            <input type="time" />
          </label>

          <label>Location:
            <input type="text" placeholder="Address" />
            <input type="text" placeholder="City" />
            <input type="state" placeholder="State" />
            <input type="zipcode" placeholder="Zipcode" />
          </label>

          <label>Price:
            <input type="text" placeholder="$ or Free" />
          </label>

          <label>Age:
            <input type="string" placeholder="Age Range" />
          </label>

          <label>Category:
            <select>
              <option value="DEFAULT" hidden>Category</option>
              <option value="sports">Sports</option>
              <option value="arts">Art</option>
              <option value="music">Music</option>
              <option value="performance">Performance</option>
              <option value="education">Educational</option>
              <option value="volunteer">Volunteer</option>
            </select>
          </label>

          <label>Description
            <textarea />
          </label>

        </fieldset>
        <button>SUBMIT</button>

      </form>
    </>
  );
}
export default SubmitEvent;
