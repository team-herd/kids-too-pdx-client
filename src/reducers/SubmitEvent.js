import {
  CREATE_EVENT,
  UPDATE_CONTACT,
  UPDATE_EVENT_NAME,
  UPDATE_DATE,
  UPDATE_LOCATION,
  UPDATE_TIME,
  UPDATE_PRICE,
  UPDATE_RRATE,
  UPDATE_MINAGE,
  UPDATE_MAXAGE,
  UPDATE_CATEGORY,
  UPDATE_DESCRIPTION
} from '../actions/submitEvent';

const initialState = {
  contact: {},
  name: '',
  date: '',
  time: '',
  location: {},
  price: '',
  rr: true,
  minAge: '',
  maxAge: '',
  category: '',
  description: ''
};

export default function reducer(state = initialState, { payload, type }) {
  switch(type) {
    case CREATE_EVENT: 
      return {
        ...state
      };
    case UPDATE_CONTACT:
      return {
        ...state,
        contact: payload
      };
    case UPDATE_EVENT_NAME:
      return {
        ...state,
        name: payload
      };
    case UPDATE_DATE:
      return {
        ...state,
        date: payload
      };
    case UPDATE_TIME:
      return {
        ...state,
        time: payload
      };
    case UPDATE_LOCATION:
      return {
        ...state,
        location: payload
      };
    case UPDATE_PRICE:
      return {
        ...state,
        price: payload
      };
    case UPDATE_RRATE:
      return {
        ...state,
        rr: payload
      };
    case UPDATE_MINAGE:
      return {
        ...state,
        minAge: payload
      };
    case UPDATE_MAXAGE:
      return {
        ...state,
        maxAge: payload
      };
    case UPDATE_CATEGORY: 
      return {
        ...state,
        category: payload
      };
    case UPDATE_DESCRIPTION: 
      return {
        ...state,
        description: payload
      };
    default:
      return state;
  }
}
