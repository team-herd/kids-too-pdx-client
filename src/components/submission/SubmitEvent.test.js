import React from 'react';
import renderer from 'react-test-renderer';
import SubmitEvent from './SubmitEvent';

describe('SubmitEvent', () => {
  it('snapshots SubmitEvent', () => {
    const props = {
      _id: 1,
      image: 'image.com',
      time: '2pm',
      name: 'House Event',
      date: new Date().toString(),
      contact: {
        contactName: 'Joe',
        email: 'joe@email.com',
        phone: '555-123-4567'
      },
      location: {
        venue: 'My House',
        address: '123 Main St.',
        city: 'Portland',
        state: 'OR',
        zipcode: 97223
      },
      price: 250,
      ageMin: 9,
      ageMax: 12,
      category: 'Art',
      description: 'The bestest event',
      rrSelect: jest.fn(),
      handleChange: jest.fn(),
      handleSubmit: jest.fn()
    };

    const tree = renderer.create(
      <SubmitEvent {...props} />
    );

    expect(tree).toMatchSnapshot();
  });
});
