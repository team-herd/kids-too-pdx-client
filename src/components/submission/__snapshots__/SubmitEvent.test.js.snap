// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitEvent snapshots SubmitEvent 1`] = `
<form
  className="Form"
  onSubmit={[Function]}
>
  <fieldset>
    <legend>
      Contact Info
    </legend>
    <label>
      Name:
      <input
        name="contactName"
        onChange={[MockFunction]}
        required={true}
        type="text"
        value="Joe"
      />
    </label>
    <label
      name="email"
    >
      Email:
      <input
        name="contactEmail"
        onChange={[MockFunction]}
        required={true}
        type="email"
        value="joe@email.com"
      />
    </label>
    <label>
      Phone #:
      <input
        name="contactPhone"
        onChange={[MockFunction]}
        required={true}
        type="tel"
        value="555-123-4567"
      />
    </label>
  </fieldset>
  <fieldset>
    <legend>
      Event Info
    </legend>
    <label>
      Name:
      <input
        name="name"
        onChange={[MockFunction]}
        type="text"
        value="House Event"
      />
    </label>
    <label>
      Date:
      <input
        name="date"
        onChange={[MockFunction]}
        required={true}
        type="date"
        value="Fri Mar 29 2019 08:53:35 GMT-0700 (Pacific Daylight Time)"
      />
    </label>
    <label>
      Time:
      <input
        name="time"
        onChange={[MockFunction]}
        required={true}
        type="time"
        value="2pm"
      />
    </label>
    <label>
      Location:
      <input
        name="locationVenue"
        onChange={[MockFunction]}
        placeholder="Venue"
        required={true}
        type="text"
        value="My House"
      />
      <input
        name="locationAddress"
        onChange={[MockFunction]}
        placeholder="Address"
        required={true}
        type="text"
        value="123 Main St."
      />
      <input
        name="locationCity"
        onChange={[MockFunction]}
        placeholder="City"
        type="text"
        value="Portland"
      />
      <input
        name="locationState"
        onChange={[MockFunction]}
        placeholder="State"
        type="state"
        value="OR"
      />
      <input
        name="locationZipcode"
        onChange={[MockFunction]}
        placeholder="Zipcode"
        type="zipcode"
        value={97223}
      />
    </label>
    <label>
      Price:
      <input
        name="price"
        onChange={[MockFunction]}
        placeholder="$"
        required={true}
        type="number"
        value={250}
      />
    </label>
    <label
      className="radio"
    >
      Reduced Rate Available:
      <input
        onChange={[MockFunction]}
        type="checkbox"
      />
    </label>
    <label>
      Age Range:
      <input
        name="minAge"
        onChange={[MockFunction]}
        placeholder="Min Age"
        required={true}
        type="number"
      />
      <input
        name="maxAge"
        onChange={[MockFunction]}
        placeholder="Max Age"
        required={true}
        type="number"
      />
    </label>
    <label>
      Category:
      <select
        name="category"
        onChange={[MockFunction]}
        required={true}
        value="Art"
      >
        <option
          hidden={true}
          value="DEFAULT"
        >
          Category
        </option>
        <option
          value="sports"
        >
          Sports
        </option>
        <option
          value="arts"
        >
          Art
        </option>
        <option
          value="music"
        >
          Music
        </option>
        <option
          value="performance"
        >
          Performance
        </option>
        <option
          value="educational"
        >
          Educational
        </option>
        <option
          value="volunteer"
        >
          Volunteer
        </option>
      </select>
    </label>
    <label>
      Description
      <textarea
        name="description"
        onChange={[MockFunction]}
        required={true}
        value="The bestest event"
      />
    </label>
  </fieldset>
  <button>
    SUBMIT
  </button>
</form>
`;
